foreshadow/intents/general.py

```diff
@@ -191,8 +200,41 @@ def column_summary(cls, df):
                 top10: top 10 most frequent values or empty array if mostly unique
                     [(value, count),...,]
         """
-        data = df.ix[:, 0]
-        nan_num = int(data.isnull().sum())
-        mode, top10 = _mode_freq(data)
 
-        return OrderedDict([("nan", nan_num), ("mode", mode), ("top10", top10)])
+        return _standard_col_summary(df)
+
+
+class TextIntent(GenericIntent):
+    """See base class.
+
+    All features can be treated as text
+
+    """
+
+    children = []
+    """No children"""
+
+    single_pipeline_template = [PipelineTemplateEntry("text", SmartText, False)]
+    """Encodes the column automatically"""
+
+    multi_pipeline_template = []
+    """No multi pipeline"""
+
+    @classmethod
+    def is_intent(cls, df):
+        """Every column can be interpreted as a text"""
+
+        return True
+
+    @classmethod
+    def column_summary(cls, df):
+        """Returns standard computed statistics for a TextIntent column
+
+            The following are computed:
+                nan: count of nans pass into dataset
+                mode: mode or np.nan if data is mostly unique
+                top10: top 10 most frequent values or empty array if mostly unique
+                    [(value, count),...,]
+        """
+
+        return _standard_col_summary(df)
```

Comment:
Is this summary adequate for text-based columns? I don't have a lot of experience with ngrams and text vectorizers but it seems there may be more useful diagnostics to get.

Reply:
I think so, I looked around and did not note any other summary statistics that might be useful for text based columns. @robocoder99 or @jichaogp if you think of any others let me know.